### DISCUSSION TITLE
TypeError: Cannot read properties of undefined (reading 'map') when using getStaticProps in a file inside the pages directory

### DISCUSSION THREAD

**Discussion by kaf-lamed-beyt** [vercel/next.js#39326] [Help] [answered]
I'm trying to fetch data from GraphQL API endpoint with the getStaticProps data-fetching method. Everything works fine in the function, even when I log the result from the API, I can see it in the terminal.

But, when I try to render the data on the page through the props, I get a reference error.

`TypeError: Cannot read properties of undefined (reading 'map')`

Here's my code snippet

on the client-side: 

```export default function BlogPage({ articles }) {
  const result = articles;

  console.log(result);

  return (
    <React.Fragment>
      <Head>
        <title>Blog</title>
      </Head>
      <Container>
        <h3>Blog</h3>
        {articles.map((post) => {
          return <p>{post.title}</p>;
        })}
      </Container>
    </React.Fragment>
  );
}```

...and here's the data-fetching method's function

```export async function getStaticProps() {
  let articles = await getArticles();

  console.log(articles);

  return {
    props: {
      articles,
    },
  };
}```

**Comment by Babbili** [comment]
Hi @kaf-lamed-beyt did you pass `articles` as a prop in the page component ?
example
```jsx
function Page({ articles }) {
    console.log('articles', articles)

   return(
       <div>
           {
               articles?.map( article => {
                   return <p key={articles.indexOf(article)}>{ article.title }</p>
               }) 
           }
       </div>
   )
}
export default Page

export async function getStaticProps() {
    let articles = await getArticles()
    
    return {
        props: { articles }   // will be passed to the page component as a prop
    }
}
```

**Reply by kaf-lamed-beyt** [reply]
Hi @Babbili, 

I did the same thing, I'm still getting `undefined` in the console. But, when I log the result to the console in the `geStaticProps` function the data is returned correctly.

**Reply by Babbili** [reply]
> Hi @Babbili, 
> 
> I did the same thing, I'm still getting `undefined` in the console. But, when I log the result to the console in the `geStaticProps` function the data is returned correctly.
> 

Is there a link to the repo, or if you'd like to share the whole code of the file

**Reply by kaf-lamed-beyt** [reply]
[Here it is](https://github.com/kaf-lamed-beyt/wbny-cms-blog)

**Reply by dhruv1710** [reply]
It worked for me !!

**Comment by kaf-lamed-beyt** [comment, marked as answer]
So, I've been able to fix this issue by using the default method of setting up a Next.js project, which is via the command below

```bash 
npx create-next-app [app-name]
```
Initially, I was trying to bootstrap the project without using the command above, so I think Next.js could not recognize my `pages` directory that's why I kept getting the `undefined` error in the console.

**Comment by kaf-lamed-beyt** [comment]
What could be the cause of this issue? 

@leerob

**Comment by minahpro** [comment]
To fix this Use optional chaining (?.) or a fallback value (||):

   ```
     <Container>
        <h3>Blog</h3>
        {articles?.map((post) => {
          return <p>{post?.title}</p>;
        })}
      </Container>
```

Read this [post](https://medium.com/@minahmmassy/react-js-cannot-read-properties-of-undefined-reading-map-causes-solutions-494a4e84ef85) it explain well common mistake happens when trying to use `.map()`
